src/orquestra/quantum/circuits/_itertools.py

```diff
@@ -178,3 +178,48 @@ def combine_measurement_counts(
         reduce(_combine_measurements, islice(measurements_it, multiplicity))
         for multiplicity in multiplicities
     ]
+
+
+def combine_bitstrings(
+    all_bitstrings: Sequence[List[str]], multiplicities: Sequence[int]
+) -> List[List[str]]:
+    """Combine (aggregate) bitstrings of the same circuits run several times.
+
+    Suppose multiplicities is a list [1, 2 ,3]. Then, the all_bitstrings should
+    be a sequence of 1+2+3=6 elements, each of them being a sequence of bitstrings.
+    For instance, all_bitstrings could be equal to:
+    [["00", "01"], ["1", "0"], ["0"], ["001", "001"], ["111", "000"], ["000"]]
+    and then the result would be:
+    [
+        ["00", "01"],
+        ["1", "0", "0],
+        ["001", "001", "111", "000", "000"]
+    ]
+
+    Args:
+        all_bitstrings: iterable of sequences containing bitstrings
```

Comment:
```suggestion
        all_bitstrings: sequence of lists containing bitstrings
```
[nitpick]: being precise about the type of all_bitstrings